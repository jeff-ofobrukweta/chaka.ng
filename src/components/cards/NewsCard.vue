<template>
    <div class="card-news">
        <img
            class="card-news__img"
            src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
            :alt="news.title"
        />
        <div class="card-news__title">
            <h5>{{ news.title | truncate(48) }}</h5>
            <!-- <h5>{{ line1 }}</h5>
      <h5 class="card-news__ellipsis">{{ line2 }}</h5> -->
        </div>
    </div>
</template>

<script>
export default {
    name: 'news-card',
    props: {
        news: {
            type: Object,
            required: true
        }
    },
    computed: {
        line1() {
            return this.news.title.substring(0, 20);
        },
        line2() {
            return this.news.title.substring(20);
        }
    }
};
</script>
