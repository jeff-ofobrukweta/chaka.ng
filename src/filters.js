import numeral from 'numeral';

numeral.register('locale', 'ng', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    ordinal(number) {
        return number === 1 ? 'er' : 'ème';
    },

    abbreviations: {
        thousand: 'K',
        million: 'M',
        billion: 'B',
        trillion: 'T'
    },
    currency: {
        symbol: '₦'
    }
});

numeral.register('locale', 'us', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    ordinal(number) {
        return number === 1 ? 'er' : 'ème';
    },

    abbreviations: {
        thousand: 'K',
        million: 'M',
        billion: 'B',
        trillion: 'T'
    },
    currency: {
        symbol: '$'
    }
});

const numberFormat = (value, currency, decimalPlaces, showFull) => {
    if (!currency) {
        if (decimalPlaces === 2) {
            if (showFull) return numeral(value).format('0,0.00');
            return numeral(value).format('0,0.00a');
        }
        if (decimalPlaces === 4) {
            if (showFull) return numeral(value).format('0,0.0000');
            if (value >= 10000) {
                return numeral(value).format('0,0.00a');
            }
            return numeral(value).format('0,0.0000a');
        }
    }
    const locale = currency.substring(0, 2).toLowerCase();
    numeral.locale(locale);
    if (value < 10000 || showFull) {
        if (decimalPlaces !== 2) return numeral(value).format('$0,0');
        return numeral(value).format('$0,0.00');
    }
    if (decimalPlaces !== 2) return numeral(value).format('$0,0a');
    return numeral(value).format('$0,0.0a');
};

const checkNumber = (value) => {
    if (value === 0 || !Number.isNaN(+value)) {
        return true;
    }
    if (value === null || value === undefined) {
        return false;
    }
    return value;
};

export default {
    kobo(value) {
        const valueCheck = checkNumber(value);
        if (valueCheck === true) {
            return value / 100;
        }
        if (!valueCheck) {
            return '-';
        }
        return value;
    },

    currency(value, currency, showFull, decimalPlaces = 2) {
        const valueCheck = checkNumber(value);
        if (valueCheck === true) {
            return numberFormat(value, currency, decimalPlaces, showFull);
        }
        if (!valueCheck) {
            return '-';
        }
        return value;
    },

    units(value, decimalPlaces = 2, showFull) {
        const valueCheck = checkNumber(value);
        if (valueCheck === true) {
            return numberFormat(value, null, decimalPlaces, showFull);
        }
        if (!valueCheck) {
            return '-';
        }
        return value;
    },

    resolveDate(value, format) {
        const date = new Date(value);
        if (typeof date === 'object' && !Number.isNaN(date.getDate())) {
            if (!format) {
                return `${date.toLocaleString('en-NG', {
                    year: 'numeric',
                    day: '2-digit',
                    month: '2-digit',
                    hour12: true
                })}`;
            }
            return `${date.toLocaleString('en-NG', {
                hour: 'numeric',
                minute: 'numeric',
                second: 'numeric',
                year: 'numeric',
                day: '2-digit',
                month: '2-digit',
                hour12: true
            })}`;
        }
        return value;
    },

    reverseDate(value) {
        const d = new Date(value);
        let month = `${d.getMonth() + 1}`;
        let day = `${d.getDate()}`;
        const year = `${d.getFullYear()}`;

        if (month.length < 2) month = `0${month}`;
        if (day.length < 2) day = `0${day}`;
        return [year, month, day].join('-');
    },

    resolveTime(value, timeOnly) {
        const date = new Date(value);
        if (typeof date === 'object' && !Number.isNaN(date.getDate())) {
            if (timeOnly) {
                return `${date.toLocaleString('en-NG', {
                    hour: '2-digit',
                    minute: '2-digit',
                    second: '2-digit',
                    hour12: true
                })}`

            }
            return `${date.toLocaleString('en-NG', {
                hour: 'numeric',
                minute: 'numeric',
                day: '2-digit',
                month: '2-digit',
                hour12: false
            })}`;
        }
        return value;
    },

    truncate(text, length, suffix = '...') {
        if (text) {
            if (text.length > length) {
                return text.substring(0, length) + suffix;
            }
        }
        return text;
    },

    capitalize(text) {
        if (text) {
            const small = text.toLowerCase()
            const temp = small.charAt(0).toUpperCase() + small.slice(1)
            return temp
        }
        return text
    }
};
