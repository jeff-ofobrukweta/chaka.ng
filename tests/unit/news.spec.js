import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import NewsCard from '@/components/cards/NewsCard.vue';
import { newsUrl } from '../unit/constants';
import filters from '../../src/filters';



/*
Inputs

Data: the component’s data
Props: the component’s props
User Interaction: user clicks a button
Lifecycle Methods: mounted(), created(), etc.
Store: state management data
Route Params: data in the route
*/



// test the news card and check for images link is valid
describe('NewsCard-vue', () => {
  let wrapper;
  let store;
  let actions;
  let mutations;
  let state;
  const localVue = createLocalVue();
  localVue.use(Vuex);

  beforeEach(() => {
    const news = {
      imageUrl: newsUrl,
      name: "news",
      title: "Boomarang Platoon Fowl"
    }
    const collection = true;
    const dummy = true;
    actions = {
      someAction: jest.fn()
    };
    mutations = {
      someMutation: jest.fn()
    };
    state = {
      key: {}
    };
    store = new Vuex.Store({
      actions,
      mutations,
      state,
    });
    wrapper = mount(NewsCard, {
      propsData: { news, collection, dummy },
      mocks: {},
      stubs: {},
      methods: {},
      store,
      localVue,
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance in the news component', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });
  it('has main-container class', () => {
    expect(wrapper.contains('.card-news')).toBe(true);
  });

  it('snapshot renders props.news when passed', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('snapshot renders props.collection when passed', () => {
    expect(wrapper).toMatchSnapshot()
  });
  it('snapshot renders props.dummy when passed', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('renders deep rendering to check for the number of img Tag', () => {
    expect(wrapper.findAll('img').length == 1)
  });

  it('test truncate filter from the vue-filters', () => {
    const news = {
      imageUrl: newsUrl,
      name: "news",
      title: "Boomarang Platoon Fowl"
    }
    expect((filters.truncate(news.name, 5).length)).toBeLessThanOrEqual(5)
  });



  // it('should redirect anonymous users to /href <anchor tag>/ when clicking on login button', () => {
  //   const anchor = wrapper.find('a')
  //   window.location.assign = jest.fn() // Create a spy
  //   anchor.trigger('click')
  //   expect(window.location.assign).toHaveBeenCalledWith('/auth/');
  // })
})


