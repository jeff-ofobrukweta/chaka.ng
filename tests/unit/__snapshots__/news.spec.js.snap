// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard-vue snapshot renders props.collection when passed 1`] = `
<div class="card-news loader">
  <div class="loader-div"></div>
</div>
`;

exports[`NewsCard-vue snapshot renders props.dummy when passed 1`] = `
<div class="card-news loader">
  <div class="loader-div"></div>
</div>
`;

exports[`NewsCard-vue snapshot renders props.news when passed 1`] = `
<div class="card-news loader">
  <div class="loader-div"></div>
</div>
`;
